modules/gapi/perf/render/gapi_render_perf_tests_inl.hpp

```diff
@@ -0,0 +1,1144 @@
+    double  transparency;
+    MatType type;
+
+    cv::Size sz;
+    std::tie(sz, rect, color, transparency, type) = GetParam();
+
+    cv::Mat y_ref_mat, uv_ref_mat;
+
+    cv::Mat y_in_gapi_mat (sz, CV_8UC1), uv_in_gapi_mat (sz / 2, CV_8UC2),
+            y_out_gapi_mat(sz, CV_8UC1), uv_out_gapi_mat(sz / 2, CV_8UC2);
+
+    create_rand_mats(sz,     CV_8UC1, y_ref_mat  , y_in_gapi_mat);
+    create_rand_mats(sz / 2, CV_8UC2, uv_ref_mat , uv_in_gapi_mat);
+
+    cv::Mat img(rect.size(), CV_8UC3, color);
+    cv::Mat alpha(rect.size(), CV_32FC1, transparency);
+    auto tl = rect.tl();
+    cv::Point org = {tl.x, tl.y + rect.size().height};
+
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::gapi::wip::draw::Prims prims;
+    prims.emplace_back(cv::gapi::wip::draw::Image{org, img, alpha});
+
+    cv::GMat y_in, uv_in, y_out, uv_out;
+    cv::GArray<cv::gapi::wip::draw::Prim> arr;
+    std::tie(y_out, uv_out) = cv::gapi::wip::draw::renderNV12(y_in, uv_in, arr);
+
+    cv::GComputation comp(cv::GIn(y_in, uv_in, arr), cv::GOut(y_out, uv_out));
+
+    // Warm-up graph engine:
+    comp.apply(cv::gin(y_in_gapi_mat, uv_in_gapi_mat, prims),
+               cv::gout(y_out_gapi_mat, uv_out_gapi_mat));
+
+    TEST_CYCLE()
+    {
+        comp.apply(cv::gin(y_in_gapi_mat, uv_in_gapi_mat, prims),
+                   cv::gout(y_out_gapi_mat, uv_out_gapi_mat));
+    }
+
+    // OpenCV code //////////////////////////////////////////////////////////////
+    {
+        // NV12 -> YUV
+        cv::Mat yuv;
+        cv::gapi::wip::draw::cvtNV12ToYUV(y_ref_mat, uv_ref_mat, yuv);
+
+        cv::Mat yuv_img;
+        cv::cvtColor(img, yuv_img, cv::COLOR_BGR2YUV);
+        blendImageRef(yuv, org, yuv_img, alpha);
+
+        // YUV -> NV12
+        cv::gapi::wip::draw::cvtYUVToNV12(yuv, y_ref_mat, uv_ref_mat);
+    }
+
+    // Comparison //////////////////////////////////////////////////////////////
+    {
+        EXPECT_EQ(0, cv::norm(y_out_gapi_mat,  y_ref_mat));
+        EXPECT_EQ(0, cv::norm(uv_out_gapi_mat, uv_ref_mat));
+    }
+
+    SANITY_CHECK_NOTHING();
+}
+
+PERF_TEST_P_(RenderTestPolylines, RenderPolylinesPerformanceBGROCVTest)
+{
+    std::vector<cv::Point> points;
+    cv::Scalar color;
+    int thick;
+    int lt;
+    int shift;
+    MatType type;
+
+    cv::Size sz;
+    std::tie(sz, points, color, thick, lt, shift, type) = GetParam();
+
+    cv::Mat gapi_mat = cv::Mat(sz, type);
+    cv::Mat ref_mat(gapi_mat);
+
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::gapi::wip::draw::Prims prims;
+    prims.emplace_back(cv::gapi::wip::draw::Poly{points, color, thick, lt, shift});
+
+    cv::GMat in;
+    cv::GArray<cv::gapi::wip::draw::Prim> arr;
+
+    cv::GComputation comp(cv::GIn(in, arr),
+                          cv::GOut(cv::gapi::wip::draw::render3ch(in, arr)));
+
+    // Warm-up graph engine:
+    comp.apply(gin(gapi_mat, prims), gout(gapi_mat));
```

Comment:
BTW,  `Computation` have `compile` method for warmp-up, don't it ?

Reply:
In this case these are similar things
Internally, a `cv::GCompiled` object is created for the given input format configuration, which then is executed on the input data immediately. `cv::GComputation` caches compiled objects produced within `apply()` -- if this method would be called next time with the same input parameters (image formats, image resolution, etc), the underlying compiled graph will be reused without recompilation. If new metadata doesn't match the cached one, the underlying compiled graph is regenerated.

Reply:
`compile()` is not a warm-up.

"Warm-up" assumes a full run so image buffers which are allocated are actually allocated and are written to. On some systems (e.g. Linux) the memory allocation is lazy and happens on actual access but not the `new`/`malloc` thing itself.

Reply:
Got it, thanks!